### DISCUSSION TITLE
git-clang-format not in the moose environment.

### DISCUSSION THREAD

**Discussion by adityaghantasala** [idaholab/moose#22393] [Q&A General] [answered]
Hello everyone. 

I noticed that the '`git-clang-format`' command/executable/script is no long in the moose environment. 
I just installed a fresh environment using the instructions from [here](https://mooseframework.inl.gov/getting_started/installation/conda.html) and I do not get `git-clang-format`. 

Is this expected ? if this is, should I change the script `install_format_hook.sh` to reflect this ? 

Thank you in advance for any pointers.

**Comment by GiudGiud** [comment]
@milljm

**Comment by milljm** [comment, marked as answer]
It looks like this may have been dropped when we became very strict in our moose-mpich package building process. You should be able to install the clang format tools by doing the following while inside the `moose` Conda environment:

```bash
mamba install clangdev
```

I'll see about getting this package back into mix when installing `moose-mpich`.

**Reply by milljm** [reply]
Correction, I think the least amount of packages to get clang-format working is by installing the following:
```bash
mamba install clang-tools
```
I'll leave the original `clangdev` in case `clang-tools` ends up not working. I came back to edit, because I am testing these package installations on a Linux machine (which we use GCC and not LLVM). By installing clangdev, a person will actually get a clang compiler. Something that you may not want/need on a Linux machine. However, I will state that by having `clangdev` installed... everything seems to still work fine.

**Reply by milljm** [reply]
Creating a PR to address this: https://github.com/idaholab/moose/pull/22420

**Comment by adityaghantasala** [comment]
Thank you for the clarification and a quick PR.  Looking forward to use the pre commit hook again !
